app/services/user_subscriptions/create.rb

```diff
@@ -0,0 +1,72 @@
+module UserSubscriptions
+  class Create
```

Comment:
This class feels a little weird to me. We have validations in the model and then we have this creation class which has more validations in it that are manually adding errors to a general Struct. It seems to be mimicking exactly what ActiveRecord is already doing for us and feels like we might not need this abstraction.

Reply:
Agreed - it's to take out some weird logic that's only needed in the controller. For example, when creating a `UserSubscription` we only need to check for the email mismatch because the request is coming from the front end. If we created one strictly on the backend (no use case for that right now 🤷‍♂️ ), that check wouldn't be necessary.

Reply:
Yeah, I totally get that. My concern here is that I would see this class and assume that every create needs to go through this class in order to include all the necessary logic rather than this class is only needed for the controller. Maybe a rename or controller helper module 🤔 I dont really have a great solution off the top of my head.

Reply:
The context was a discussion @atsmith813 and I had about avoiding conditional validations/callbacks etc. on models, as they can be hard to reason about and add cruft to a model despite only being used in one specific circumstance.

I do agree though that `Create` by itself is a bit misleading, as it wrongly suggests that this is the canonical way of creating `UserSubscription` instances. I tend to favor relatively explicit and verbose names for services, e.g. `UserSubscriptions::CreateFromXyzForm` or `UserSubscriptions::CreateFromUserInteraction`, etc.

Reply:
I renamed it to `UserSubscriptions::CreateFromControllerParams`